File: assets/scenarios/gen_grid.py
name = 'dam_miles'

h = 0.02
s = h / 2

min_x = 0.20
max_x = 0.40
min_y = 0.20
max_y = 0.40
min_z = 0.20
max_z = 0.40
width = max_x - min_x
height = max_y - min_y
depth = max_z - min_z

p_x = int(round(width / s)) + 1
p_y = int(round(height / s)) + 1
p_z = int(round(depth / s)) + 1

p = p_x * p_y * p_z
m = 1.0
r = s

f = open(name+'.in', 'w')
f.write('{}\n'.format(p))

for x in range(p_x):
    for y in range(p_y):
        for z in range(p_z):
            f.write("{} {} {} {} {} {} {} {}\n".format(
                m, s, min_x + x * s, min_y + y * s, min_z + z * s, 0, 0, 0))

f.close()

f = open(name+'.par', 'w')
f.write("part_input_file       {}\n".format(name+'.in'))
f.write("time_end              {}\n".format(10.0))
f.write("timestep_length       {}\n".format(0.01))
f.write("x_min                 {}\n".format(0))
f.write("x_max                 {}\n".format(0.5))
f.write("y_min                 {}\n".format(0))
f.write("y_max                 {}\n".format(0.5))
f.write("z_min                 {}\n".format(0))
f.write("z_max                 {}\n".format(0.5))
f.write("x_n                   {}\n".format(int(round((0.5-0.0)/h))))
f.write("y_n                   {}\n".format(int(round((0.5-0.0)/h))))
f.write("z_n                   {}\n".format(int(round((0.5-0.0)/h))))
f.write("density               {}\n".format(p * (width * height * depth)))
f.close()

File: assets/scenarios/gen_grid.js
var fs = require('fs');
var stream = fs.createWriteStream("debug.in");

var min_x = 0.05;
var max_x = 0.45;
var min_y = 0.05;
var max_y = 0.85;
var min_z = 0.05;
var max_z = 0.85;
var w = max_x - min_x;
var h = max_y - min_y;
var d = max_z - min_z;

console.log(w);
console.log(h);
console.log(d);

var p = 100000;
var p_x = Math.round(Math.pow((p * Math.pow(w,2)) / (h * d), 1/3));
var p_y = Math.round(p_x * (h/w));
var p_z = Math.round(p_x * (d/w));

console.log(p_x);
console.log(p_y);
console.log(p_z);

var s_x = w / (p_x - 1);
var s_y = h / (p_y - 1);
var s_z = d / (p_z - 1);

console.log(s_x);
console.log(s_y);
console.log(s_z);

var radius = Math.min(s_x,s_y,s_z) / 10;

var mass = ((w * h * d) * 1000) / p;

console.log(mass);

var precision = 10 * 5;

stream.once('open', function(fd) {
    for (var x = min_x; x < max_x; x+= s_x) {
        for (var y = min_y; y < max_y; y += s_y) {
            for (var z = min_z; z < max_z; z += s_z) {
                stream.write(mass+' ' + radius + ' ' + x + ' ' + y + ' ' + z +' 0 0 0\n');
            }
        }
    }
});